src/dft.cpp

```diff
@@ -1375,4 +1375,91 @@ void fields::get_mode_mode_overlap(void *mode1_data, void *mode2_data, dft_flux
   get_overlap(mode1_data, mode2_data, flux, 0, overlaps);
 }
 
+std::vector<int> fields::get_corners(dft_fields fdft, component c){ // get the minimum and maximum ivec values of the dft monitor
+  volume *where = &v;
```

Comment:
There's a bit of subtlety when calculating the proper `ivec` corners during the DFT chunk creation that doesn't seem to be accounted for here.

If you look inside the `loop_in_chunks` routine (which is what determines the actual `ivec` corners) you notice there is some "snapping" that occurs, and that snapping depends on whether or not we are interpolating to the center of the voxel (e.g. yee_grid=False):

https://github.com/NanoComp/meep/blob/bc58782c0d16e6d59937028cbe2ffa402f5f24b0/src/loop_in_chunks.cpp#L361-L369

There's some *additional* logic below that to account for symmetries.

It's probably best to refactor *that* part of `loop_in_chunks` as `get_corners` so that we are consistently pulling the same corners.

Reply:
Steven suggested me to refactor the code from `fields::process_dft_component` in [dft.cpp](https://github.com/NanoComp/meep/blob/bc58782c0d16e6d59937028cbe2ffa402f5f24b0/src/dft.cpp#L1028). So the function `get_corners` has some code from `fields::process_dft_component` in [dft.cpp](https://github.com/NanoComp/meep/blob/bc58782c0d16e6d59937028cbe2ffa402f5f24b0/src/dft.cpp#L1028), but not from `fields::loop_in_chunks` in [loop_in_chunks.cpp](https://github.com/NanoComp/meep/blob/bc58782c0d16e6d59937028cbe2ffa402f5f24b0/src/loop_in_chunks.cpp#L340).

In my tests, `get_corners` could distinguish `self.yee_grid==False` from `True`. I have noticed the following things as expected:
1) The output of `get_corners` varies when `self.yee_grid` is changed.
2) The output of `get_corners` varies when the field component is changed in the case of `self.yee_grid==True`.
3) The output of `get_corners` does not vary when the field component is changed in the case of `self.yee_grid==False`.

In addition, after comparing the values of these outputs, I could infer that `get_corners` works on centers of voxels. (If `get_corners` does not work on centers of voxels when `self.yee_grid==False`, `get_corners` probably does not work correctly when `self.yee_grid==True`.)

Reply:
Do you think we can keep this function as it is?

Reply:
If it passes tests, then I see no reason to change it. Just note that the `dft` code *does* call `loop_in_chunks` when determining the boundaries (not just when computing the actual `dtft`), so we want to make sure we are always consistent.